server/utils.js

```diff
@@ -632,36 +678,51 @@ function getOrgReservations(franchise, listOfRestaurants, listOfReservations) {
   return objectToArray(listOfReservations).filter(
     function(element) {
       return listOfRestaurants.some(function(restaurant) {
-        return restaurant.id === element.reservationFor.name;
+        return restaurant.name === element.reservationFor;
       });
     }
   );
 }
 
-function getListByType(type, element, headers) {
+function getListByType(type, element, headers, raw) {
   var uri = '/v2/entities';
+  var options = {'type': type,'limit': '1000'};
   if (element) {
     uri += '/' + encodeURIComponent(element);
   }
+
+  if (typeof raw === 'undefined') {
+    options.options = 'keyValues';
+  }
   return authRequest(
     uri,
     'GET',
-    {'type': type,'limit': '1000'},
+    options,
     headers
   );
 }
 
-function sendRequest(method, body, identifier, headers) {
+function sendRequest(method, body, identifier, headers, payload) {
   var uri = '/v2/entities';
   if (identifier) {
     uri += '/' + encodeURIComponent(identifier);
   }
-  return authRequest(
-    uri,
-    method,
-    body,
-    headers
-  );
+  if (typeof payload === 'undefined') {
+    return authRequest(
+      uri,
+      method,
+      body,
+      headers
+    );
+  } else {
+    return authRequest(
+      uri,
+      method,
+      body,
+      headers,
+      payload
+    );
```

Comment:
strange, strange

simply check the payload argument at the called function and you will save precious lines of code